v2/pkg/engine/datasource/graphql_datasource/graphql_datasource_test.go

```diff
@@ -9101,12 +9105,157 @@ func TestSource_Load(t *testing.T) {
 			input, err = httpclient.SetUndefinedVariables(input, undefinedVariables)
 			assert.NoError(t, err)
 
-			require.NoError(t, src.Load(ctx, input, buf))
+			require.NoError(t, src.Load(ctx, input, []httpclient.File(nil), buf))
 			assert.Equal(t, `{"variables":{"b":null}}`, buf.String())
 		})
 	})
 }
 
+type ExpectedFile struct {
+	Name string
+	Size int64
+}
+
+type ExpectedRequest struct {
+	Operations string
+	Map        string
+	Files      []ExpectedFile
+}
+
+func verifyMultipartRequest(t *testing.T, r *http.Request, expected ExpectedRequest) {
+	err := r.ParseMultipartForm(10 << 20)
+	require.NoError(t, err)
+
+	for key, values := range r.MultipartForm.Value {
+		switch key {
+		case "operations":
+			assert.Equal(t, expected.Operations, values[0])
+		case "map":
+			assert.Equal(t, expected.Map, values[0])
+		}
+	}
+
+	for i, expectedFile := range expected.Files {
+		values, exists := r.MultipartForm.File[strconv.Itoa(i)]
+		if !exists {
+			t.Fatalf("expected file %s not found in MultipartForm.File", expectedFile.Name)
+		}
+		assert.Equal(t, values[0].Filename, expectedFile.Name)
+		assert.Equal(t, values[0].Size, expectedFile.Size)
+	}
+}
+
+func TestLoadFiles(t *testing.T) {
+	t.Run("single file", func(t *testing.T) {
+		queryString := `mutation($file: Upload!){singleUpload(file: $file)}`
+		variableString := `{"file":null}`
```

Comment:
Can you explain why file us null?

Reply:
Hey @thisisnithin , please let me jump and explain this one. @jensneuse This is the usual thing in GQL libraries. We bring this from our previous experience with file uploads without federation. The appollo client generate requests like this.